src/Application/Commands/PushDailyFundTotals.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace MatchBot\Application\Commands;
+
+use MatchBot\Application\Messenger\FundTotalUpdated;
+use MatchBot\Domain\FundRepository;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\RoutableMessageBus;
+
+/**
+ * @see RetrospectivelyMatch which does a similar thing for just-closed campaigns as one of the last campaign close tasks.
+ */
+#[AsCommand(
+    name: 'matchbot:push-daily-fund-totals',
+    description: 'Pushes champion funds used totals to Salesforce, for all associated with open campaigns'
+)]
+class PushDailyFundTotals extends LockingCommand
+{
+    public function __construct(
+        private readonly FundRepository $fundRepository,
+        private readonly LoggerInterface $logger,
+        private readonly RoutableMessageBus $bus,
+    ) {
+        parent::__construct();
+    }
+
+    public function doExecute(InputInterface $input, OutputInterface $output): int
+    {
+        $funds = $this->fundRepository->findForCampaignsOpenAt(new \DateTime('now'));
+        foreach ($funds as $fund) {
+            if ($fund->getFundType()->isPledge()) {
+                // Skip pledges from daily total push as Salesforce doesn't yet do anything with that info
+                // and it could be a lot of requests against a volume-limited Site.
+                continue;
+            }
+
+            $this->bus->dispatch(new Envelope(FundTotalUpdated::fromFund($fund)));
```

Comment:
This is going to cause this function to run once for each non-pledge fund: https://github.com/thebiggive/matchbot/blob/12a8c1fe42d721dad72ee50a29f640266176ac19/src/Client/Fund.php#L49

which I think will then run the `setUsedAmount` function for each one in SF.

I'm not sure exactly what the limitation in SF is, but I wonder if we could make it a lot more efficient if we had a function in SF we could use that would set the usedAmount for many funds, and send it a map from fund id to amount. Probably something to think about for deploying after GMF, but if that works could that mean we update the funds much more often than once a day?

Reply:
Certainly an option to build but I don't know if the total bandwidth would be that different or if there's a need to have this more often than daily and on close, nobody has mentioned a more frequent need.